compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -310,21 +310,46 @@ OMR::Z::TreeEvaluator::dsqrtEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
    TR::Node * firstChild = node->getFirstChild();
    TR::Register * targetRegister = NULL;
-   TR::Register * opRegister = cg->evaluate(firstChild);
 
-   if (cg->canClobberNodesRegister(firstChild))
+
+   // Calculate it for ourselves
+   if (firstChild->getOpCode().isLoadConst())
       {
-      targetRegister = opRegister;
+      union { double valD; int64_t valI; } result;
+      targetRegister = cg->allocateRegister(TR_FPR);
+      result.valD = sqrt(firstChild->getDouble());
+      TR::S390ConstantDataSnippet * cds = cg->findOrCreate8ByteConstant(node, result.valI);
+      generateRXInstruction(cg, TR::InstOpCode::LD, node, targetRegister, generateS390MemoryReference(cds, cg, 0, node));
       }
    else
       {
-      targetRegister = cg->allocateRegister(TR_FPR);
+      TR::Register * opRegister = NULL;
+
+      //See whether to use SQDB or SQDBR depending on how many times it is referenced
+      if (firstChild->isSingleRefUnevaluated() && firstChild->getOpCodeValue() == TR::dloadi)
+         {
+         targetRegister = cg->allocateRegister(TR_FPR);
+         generateRXEInstruction(cg, TR::InstOpCode::SQDB, node, targetRegister, generateS390MemoryReference(firstChild, cg), 0);
+         }
+      else
+         {
+         opRegister = cg->evaluate(firstChild);
+
+         if (cg->canClobberNodesRegister(firstChild))
```

Comment:
This query is unnecessary. Even if `canClobberNodesRegister` returns true, reusing the `opRegister` as the `targetRegister` will not make a difference in the binary encoding because `canClobberNodesRegister` effectively asks the question "is this the end of the live range of this virtual register". If the answer to this question is yes, then the register allocator is free to reuse the same register across the `SQDBR` instruction since the first use of `targetRegister` is within the `SQDBR` instruction itself. We should remove this piece of code and simply allocate a virtual register and use it.